letsencrypt/certbot/tasks/main.yml

```diff
@@ -11,6 +11,35 @@
     state: present
   tags: install
 
+- name: Create a certbot.service override directory
+  file:
+    owner: root
+    group: root
+    mode: 0755
+    path: /etc/systemd/system/certbot.service.d
+    state: directory
+  tags: configure
+  when: proxy_host is defined and proxy_port is defined
+
+- name: set certbot's proxy values
+  community.general.ini_file:
+    dest: /etc/systemd/system/certbot.service.d/override.conf
+    owner: root
+    group: root
+    mode: 0644
+    ignore_spaces: true
+    no_extra_spaces: true
+    section: Service
+    option: Environment
+    value: "http_proxy=http://{{proxy_host}}:{{proxy_port}} https_proxy=http://{{proxy_host}}:{{proxy_port}}"
+  tags: configure
+  when: proxy_host is defined and proxy_port is defined
+
+- name: make systemd reread configs
+  ansible.builtin.systemd_service:
```

Comment:
it would be nice if this would be conditional based on changed attribute from previous task

Reply:
lemme have a look

Reply:
had a look, I think it should do now. is that what you had in mind?